snuba/clickhouse/formatter.py

```diff
@@ -60,6 +62,11 @@ def visitLiteral(self, exp: Literal) -> str:
             return escape_string(exp.value)
         elif isinstance(exp.value, (int, float)):
             return str(exp.value)
+        elif isinstance(exp.value, datetime):
+            value = exp.value.replace(tzinfo=None, microsecond=0)
+            return "toDateTime('{}')".format(value.isoformat())
+        elif isinstance(exp.value, date):
+            return "toDate('{}')".format(exp.value.isoformat())
```

Comment:
This is because there is some dataset specific logic that parses string literal containing dates into datetime objects. I am not sure why nor whether we need it. But I am keeping this logic at least for now in order not to diverge from the old behavior.